avm/res/network/virtual-network-gateway/main.bicep

```diff
@@ -155,10 +164,36 @@ var virtualGatewayPipNameVar = isActiveActiveValid
 var vpnTypeVar = gatewayType != 'ExpressRoute' ? vpnType : 'PolicyBased'
 
 var isBgpValid = gatewayType != 'ExpressRoute' ? enableBgp : false
-var bgpSettings = {
+
+// Potential configurations (active-active vs active-passive)
+var bgpSettings = isActiveActiveValid
+? {
+  asn: asn
+  bgpPeeringAddresses: [
```

Comment:
This is being discussed offline, next step is on me

Reply:
@fabmas as discussed offline, in this scenario the usage of UDT discriminator seems appropriate 
Ref: https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/user-defined-data-types#declare-tagged-union-type

We can try using the activeActive property as discriminator, requiring a second pip and bgpPeeringAddresses if enabled.

Please investigate on it and update the PR using that logic so that the review can continue based on it.